.github/workflows/ts_tests.yml

```
@@ -0,0 +1,22 @@
+name: Tree-sitter test
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+    name: Tree-sitter grammar tests
+    runs-on: ubuntu-latest
+    steps:
+        - name: Checkout repository
+          uses: actions/checkout@v4
+
+        - run: rustup toolchain install stable
```

I would prefer to use an action for this:
```suggestion
        - name: Install Rust Toolchain
          uses: dtolnay/rust-toolchain@stable
```